tslib/src/functions/gias-sync/gias-orchestrator.ts

```diff
@@ -0,0 +1,32 @@
+import { IEstablishmentFilter, MtcEstablishmentFilter } from './establishment-type-age.filter'
+import { GiasExtractParser, IGiasExtractParser } from './gias-extract-parser'
+import { GiasWebService, IGiasWebService } from './web/gias-web.service'
+
+export class GiasOrchestrator {
+
+  private giasWebService: IGiasWebService
+  private extractParser: IGiasExtractParser
+  private estabFilter: IEstablishmentFilter
+
+  constructor (giasWebService?: IGiasWebService, extractParser?: IGiasExtractParser, estabFilter?: IEstablishmentFilter) {
+    if (giasWebService === undefined) {
+      giasWebService = new GiasWebService()
+    }
+    this.giasWebService = giasWebService
+    if (extractParser === undefined) {
+      extractParser = new GiasExtractParser()
+    }
+    this.extractParser = extractParser
+    if (estabFilter === undefined) {
+      estabFilter = new MtcEstablishmentFilter()
+    }
+    this.estabFilter = estabFilter
+  }
+
+  async execute (): Promise<any> {
+    const extract = await this.giasWebService.GetExtract(12345)
```

Comment:
Why does this method start with an uppercase letter? Odd.

Reply:
probably hangover from the raw soap call, which is uppercase. updated